funnel/views/project.py

```diff
@@ -110,16 +111,24 @@ def view(self):
         rsvp_form = RsvpForm(obj=self.obj.rsvp_for(g.user))
         transition_form = ProjectTransitionForm(obj=self.obj)
         schedule_transition_form = ProjectScheduleTransitionForm(obj=self.obj)
+        project_save_form = ProjectSaveForm()
+        project_currently_saved = self.obj.is_saved_by(current_auth.user)
         return {'project': self.obj,
             'rsvp_form': rsvp_form, 'transition_form': transition_form,
-            'schedule_transition_form': schedule_transition_form}
+            'schedule_transition_form': schedule_transition_form,
+            'project_save_form': project_save_form,
+            'project_currently_saved': project_currently_saved}
 
     @route('proposals')
     @render_with('proposals.html.jinja2')
     @requires_permission('view')
     def view_proposals(self):
         cfp_transition_form = ProjectCfpTransitionForm(obj=self.obj)
-        return {'project': self.obj, 'cfp_transition_form': cfp_transition_form}
+        project_save_form = ProjectSaveForm()
+        project_currently_saved = self.obj.is_saved_by(current_auth.user)
```

Comment:
Instead of repeating this code, move it into a method in the class. Say, `self.project_details()`, which gives you a dict that you can add more attributes to before returning to `render_with`.

Reply:
You could also delegate this to the template. It can call `current_view.save_details()` if and only if required, saving you some computation.

Reply:
This still applies -- SavedProjectForm is required by every view under Project, as the save button will appear in all of them (in desktop view).